core/src/main/java/com/graphhopper/reader/dem/BridgeElevationInterpolator.java

```diff
@@ -29,13 +30,16 @@
  */
 public class BridgeElevationInterpolator extends AbstractEdgeElevationInterpolator {
 
+    private final int bridgeRef;
+
     public BridgeElevationInterpolator(GraphHopperStorage storage, StringEncodedValue roadEnvironmentEnc) {
         super(storage, roadEnvironmentEnc);
+        //TODO We should not hardcode this string here
+        bridgeRef = roadEnvironmentEnc.indexOf("bridge");
     }
 
     @Override
     protected boolean isInterpolatableEdge(EdgeIteratorState edge) {
-        //TODO We should not hardcode this string here
-        return edge.get(roadEnvironmentEnc).equals("bridge");
+        return edge.get((IntEncodedValue) roadEnvironmentEnc) == bridgeRef;
```

Comment:
It is important to move the cast into the constructor to avoid performance issues.